src/language-js/printer-estree.js

```diff
@@ -639,7 +643,11 @@ function printPathNoParens(path, options, print, args) {
         parent.type === "Property";
 
       const samePrecedenceSubExpression =
-        isBinaryish(n.left) && shouldFlatten(n.operator, n.left.operator);
+        isBinaryish(n[leftNodeName]) &&
+        shouldFlatten(
+          operator,
```

Comment:
As you can see below, this is used only for logical expressions, so we can rewrite this part into an `if`:
```js
if (!shouldNotIndent) {
  if (shouldInlineLogicalExpression(n)) {
    const samePrecedenceSubExpression = isBinaryish(n.left) && shouldFlatten(n.operator, n.left.operator);
    shouldNotIndent = !samePrecedenceSubExpression;
  } else {
    shouldNotIndent = shouldIndentIfInlining;
  }
}
```
These variable names (`shouldNotIndent`, `shouldIndentIfInlining`) look really confusing. 🤔